### PR TITLE
[Fix](partial update) Fix core when doing partial update on tables with row column after schema change

### PR DESCRIPTION
**Pull request by bobhan1**
## Proposed changes

When reading values from old rows during partial update, we should use the schema from the current load rather than the rowset's schema to avoid core if the rowset's schema is different from the load's schema due to schema change.

img_v3_0251_f02e8efa-b693-4d92-a75c-4b39214a6f9g

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by bobhan1** [comment]
run buildall

**Comment by bobhan1** [comment]
run buildall

**Comment by bobhan1** [comment]
run buildall

File: be/src/olap/tablet.h [hunk 1]
```diff
@@ -438,7 +438,8 @@ class Tablet final : public BaseTablet {
                                  const TabletColumn& tablet_column,
                                  vectorized::MutableColumnPtr& dst);
 
-    Status fetch_value_through_row_column(RowsetSharedPtr input_rowset, uint32_t segid,
+    Status fetch_value_through_row_column(RowsetSharedPtr input_rowset,
```
Comment (zhannngchen) [inline be/src/olap/tablet.h:441]: Add comments on this methods to explain some critical parameters, emphasize that `cids` should be consistent with `tablet_schema`.

**Comment by bobhan1** [comment]
run buildall

**Review by zhannngchen — APPROVED** [review]
LGTM

**Review by liaoxin01 — APPROVED** [review]
LGTM

**Pull request merged**